services/skipchain/skipchain.go

```diff
@@ -0,0 +1,321 @@
+package skipchain
+
+import (
+	"crypto/rand"
+	"errors"
+
+	"fmt"
+
+	"bytes"
+
+	"sync"
+
+	"github.com/dedis/cothority/lib/cosi"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+// ServiceName can be used to refer to the name of this service
+const ServiceName = "Skipchain"
+
+func init() {
+	sda.RegisterNewService(ServiceName, newSkipchainService)
+	skipchainSID = sda.ServiceFactory.ServiceID(ServiceName)
+}
+
+var skipchainSID sda.ServiceID
+
+// Service handles adding new SkipBlocks
+type Service struct {
+	*sda.ServiceProcessor
+	// SkipBlocks points from SkipBlockID to SkipBlock but SkipBlockID is not a valid
+	// key-type for maps, so we need to cast it to string
+	SkipBlocks map[string]*SkipBlock
+	sbMutex    sync.Mutex
+	path       string
+}
+
+// ProposeSkipBlock takes a hash for the latest valid SkipBlock and a SkipBlock
+// that will be verified. If the verification returns true, the new SkipBlock
+// will be signed and added to the chain and returned.
+// If the given nil as the latest block it verify if we are actually creating
+// the first (genesis) block and create it. If it is called with nil although
+// there already exist previous blocks, it will return an error.
+func (s *Service) ProposeSkipBlock(e *network.Entity, psbd *ProposeSkipBlock) (network.ProtocolMessage, error) {
+	prop := psbd.Proposed
+	var latest *SkipBlock
+	if !psbd.LatestID.IsNull() {
+		var ok bool
+		latest, ok = s.getSkipBlockByID(psbd.LatestID)
+		if !ok {
+			return nil, errors.New("Didn't find latest block")
+		}
+		prop.MaximumHeight = latest.MaximumHeight
+		prop.ParentBlockID = latest.ParentBlockID
+		prop.VerifierID = latest.VerifierID
+		if s.verifyNewSkipBlock(latest, prop) {
+			s.updateSkipBlockLinks(latest, prop)
+		} else {
+			return nil, errors.New("Verification error")
+		}
+		if err := s.startPropagation(latest); err != nil {
+			return nil, err
+		}
+	} else {
+		// TODO: allow for other verificators
```

Comment:
How do you envision such verificators system ? Using some kind of registration like ProtocolInstance and Service, stg like: ` RegisterSkipchainVerificators( ... )` ?
Don't you think it makes sense to implement that in this PR ? We'll likely need it very soon if we want to start using it, no ?